misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -12,21 +12,107 @@ export const pickingReducer = ({ draftState, action }) => {
       return reduceOnUpdateQtyPicked(draftState, action.payload);
     }
 
+    case types.UPDATE_ALT_PICKING_STEP_QTY: {
+      return reduceOnAlternatePickingQty(draftState, action.payload);
+    }
+
     default: {
       return draftState;
     }
   }
 };
 
+const reduceOnAlternatePickingQty = (draftState, payload) => {
+  const { wfProcessId, activityId, lineId, stepId, altStepId, scannedHUBarcode, qtyPicked, qtyRejectedReasonCode } =
+    payload;
+
+  const draftWFProcess = draftState[wfProcessId];
+  const draftAltStep =
+    draftWFProcess.activities[activityId].dataStored.lines[lineId].steps[stepId].altSteps.genSteps[altStepId];
+  draftAltStep.scannedHUBarcode = scannedHUBarcode;
+  draftAltStep.qtyPicked = qtyPicked;
+  draftAltStep.qtyRejectedReasonCode = qtyRejectedReasonCode;
+
+  return draftState;
+};
+
+const generateAlternativeSteps = ({ draftState, wfProcessId, activityId, lineId, stepId, qtyRejected }) => {
```

Comment:
IMHO think `qtyToAllocate` is what we need and not `qtyRejected`.

Reply:
also, i think before blindly re-initializing the genSteps array, 
you shall consider what was recorded there before.

case: user is reporting on alternative step (qty=1) and then to another alternative (qty=2).
In this case i want to have both alternative steps with qty=1 and qty=2 respectively.
And the total qty picked is 1+2=3

Reply:
also, it would be nice considering unit tests for functions like this one.